backend/src/Edna/Library/DB/Query.hs

```diff
@@ -96,16 +125,32 @@ getCompoundById (SqlId compoundId) = runSelectReturningOne' $ select $ do
 getCompounds :: Edna [CompoundRec]
 getCompounds = runSelectReturningList' $ select $ all_ $ esCompound ednaSchema
 
-editCompoundChemSoft :: SqlId 'CompoundId -> Text -> Edna ()
+editCompoundChemSoft :: SqlId 'U.CompoundId -> Text -> Edna ()
 editCompoundChemSoft (SqlId compoundId) link = runUpdate' $ update (esCompound ednaSchema)
   (\c -> cChemsoftLink c <-. val_ (Just link))
   (\c -> cCompoundId c ==. val_ (SqlSerial compoundId))
 
+getCompoundByName :: Text -> Edna (Maybe CompoundRec)
+getCompoundByName name = runSelectReturningOne' $ select $ do
+  compounds <- all_ $ esCompound ednaSchema
+  guard_ (LDB.cName compounds ==. val_ name)
+  pure compounds
+
+insertCompound :: Text -> Edna (Maybe (Text, Word32))
+insertCompound compoundName = (compoundName,) . unSerial . cCompoundId <<$>> do
+  runInsert' $ Pg.insert
+    (esCompound ednaSchema)
+    (insertExpressions [CompoundRec default_ (val_ compoundName) default_ default_])
+    (Pg.onConflict (Pg.conflictingFields cName) Pg.onConflictDoNothing)
+  getCompoundByName compoundName
+
+--------------------------
+-- Test methodology
+--------------------------
+
 getMethodologyById :: SqlId 'MethodologyId -> Edna (Maybe TestMethodologyRec)
-getMethodologyById (SqlId methodologyId) = runSelectReturningOne' $ select $ do
-  methodologies <- all_ $ esTestMethodology ednaSchema
-  guard_ (tmTestMethodologyId methodologies ==. val_ (SqlSerial methodologyId))
-  pure methodologies
+getMethodologyById (SqlId methodologyId) = runSelectReturningOne' $
+  lookup_ (esTestMethodology ednaSchema) $ TestMethodologyId $ fromIntegral methodologyId
```

Comment:
AFAIU, here `fromIntegral` just applies `SqlSerial` constructor. If that's correct, I think changing `fromIntegral` to `SqlSerial` will make this code more descriptive and shorter at the same time.